src/components/ScaleLattice.vue

```diff
@@ -1,19 +1,195 @@
+    throw new Error("Points must have the same dimensionality");
+  }
+
+  let sumOfSquares = 0;
+  for (let i = 0; i < point1.length; i++) {
+    sumOfSquares += Math.pow(point2[i] - point1[i], 2);
+  }
+
+  return Math.sqrt(sumOfSquares);
+}
+
+// === Computed state ===
+const equave = 2; // TODO unassume equave is 2
+const edges = computed(() => {
+  const combinations = getPairCombinations(props.scale.intervals);
+  return combinations
+    .map(([a, b]) => {
+      return {
+        diff: euclideanDistance(
+          a.monzo.vector.valueOf().slice(1),
+          b.monzo.vector.valueOf().slice(1)
+        ),
+        pair: [a.name, b.name],
+      };
+    })
+    .filter(({ diff }) => diff === 1);
+});
+
+const nodes = computed(() => {
+  return props.scale.intervals.map(({ monzo, name: label }) => {
+    const node: Node = monzo.vector.reduce(
+      ({ x, y }, item, i) => {
+        if (PRIMES.indexOf(i) === equave) {
+          return { x, y, label };
+        }
+
+        const componentVal = item.valueOf();
+        if (componentVal === 0) {
+          return { x, y, label };
+        } else {
+          const p: string = PRIMES[i].toString();
+          const vector = coords[p] || { x: 0, y: 0 };
+          return {
+            x: x + vector.x * componentVal,
+            y: y + vector.y * componentVal,
+            label,
+          };
+        }
+      },
+      { x: 0, y: 0, label: "" }
+    );
+    return node;
+  });
+});
+
+const nodesByLabel = computed(() =>
+  nodes.value.reduce((acc, node) => ({ ...acc, [node.label]: node }), {})
+);
+
+const viewBox = computed(() =>
+  nodes.value.reduce(
+    ({ minX, maxX, minY, maxY }, { x, y }) => {
+      return {
+        minX: minX < x ? minX : x - 20,
+        minY: minY < y ? minY : y - 20,
+        maxX: maxX > x ? maxX : x + 20,
+        maxY: maxY > y ? maxY : y + 20,
+      };
+    },
+    { minX: 0, minY: 0, maxX: 0, maxY: 0 }
+  )
+);
 </script>
 
 <template>
-  <canvas ref="canvas" :width="width" :height="height"> </canvas>
+  <div>
+    <svg
+      :width="props.width"
+      :height="props.height"
+      xmlns="http://www.w3.org/2000/svg"
+      style="background-color: transparent"
+      :viewBox="
+        [
+          viewBox.minX,
+          viewBox.minY,
+          viewBox.maxX - viewBox.minX,
+          viewBox.maxY - viewBox.minY,
+        ].join(' ')
+      "
+    >
+      <!-- Top-left to top-right line -->
+      <line
+        v-for="edge of edges"
+        :key="edge.pair.join()"
+        :x1="nodesByLabel[edge.pair[0]].x"
+        :x2="nodesByLabel[edge.pair[1]].x"
+        :y1="nodesByLabel[edge.pair[0]].y"
+        :y2="nodesByLabel[edge.pair[1]].y"
+        stroke="white"
+        stroke-width="1"
+      />
+      <g v-for="node of nodes" :key="node.label">
+        <circle :cx="node.x" :cy="node.y" r="4" fill="white" />
+        <text
+          :x="node.x + 2"
+          :y="node.y - 4"
+          font-family="Arial"
```

Comment:
Stuff like fonts and fill colors should be in a `<style>` section and the elements should only get a class to reference the styles.